server/routes/activities/suspensions/handlers/suspendUntil.test.ts

```diff
@@ -0,0 +1,252 @@
+
+      expect(errors).toEqual(expect.arrayContaining([{ property: 'datePresetOption', error: 'Select a date' }]))
+    })
+
+    it('date preset option must be a known value', async () => {
+      const body = {
+        datePresetOption: 'unknown',
+      }
+
+      const requestObject = plainToInstance(SuspendUntil, body)
+      const errors = await validate(requestObject).then(errs => errs.flatMap(associateErrorsWithProperty))
+
+      expect(errors).toEqual(expect.arrayContaining([{ property: 'datePresetOption', error: 'Select a date' }]))
+    })
+
+    it('tomorrow cant be selected if the allocation ends today', async () => {
+      const body = {
+        datePresetOption: 'tomorrow',
+      }
+
+      const requestObject = plainToInstance(SuspendUntil, {
+        ...body,
+        suspendJourney: { earliestAllocationEndDate: formatDate(new Date(), 'yyyy-MM-dd') },
+      })
+      const errors = await validate(requestObject, {}).then(errs => errs.flatMap(associateErrorsWithProperty))
+
+      expect(errors).toEqual(
+        expect.arrayContaining([
+          {
+            property: 'datePresetOption',
+            error: `Suspension must be ended on or before the allocation end date ${formatDate(new Date(), 'dd/MM/yyyy')}`,
+          },
+        ]),
+      )
+    })
+
+    it('other date must be a valid date', async () => {
+      const body = {
+        datePresetOption: 'other',
+        date: '04/13/2030',
+      }
+
+      const requestObject = plainToInstance(SuspendUntil, {
+        ...body,
+        suspendJourney: {
+          earliestAllocationStartDate: formatDate(new Date('2030-05-05'), 'yyyy-MM-dd'),
+          earliestAllocationEndDate: formatDate(new Date('2030-05-10'), 'yyyy-MM-dd'),
+        },
+      })
+      const errors = await validate(requestObject, {}).then(errs => errs.flatMap(associateErrorsWithProperty))
+
+      expect(errors).toEqual(
+        expect.arrayContaining([
+          {
+            property: 'date',
+            error: 'Enter a valid date',
+          },
+        ]),
+      )
+    })
+
+    it('cannot enter a date before today', async () => {
+      const body = {
+        datePresetOption: 'other',
+        date: '04/12/2023',
+      }
+
+      const requestObject = plainToInstance(SuspendUntil, {
+        ...body,
+        suspendJourney: {
+          earliestAllocationStartDate: formatDate(new Date('2020-05-05'), 'yyyy-MM-dd'),
+          earliestAllocationEndDate: formatDate(new Date('2030-05-10'), 'yyyy-MM-dd'),
+        },
+      })
+      const errors = await validate(requestObject, {}).then(errs => errs.flatMap(associateErrorsWithProperty))
+
+      expect(errors).toEqual(
+        expect.arrayContaining([
+          {
+            property: 'date',
+            error: "Enter a date on or after today's date",
+          },
+        ]),
+      )
+    })
+
+    it('other date must not be lower than between the start and end dates of the allocation', async () => {
```

Comment:
Either my eyes are not reading correctly (I had a late night) or the test description reads a little odd ...